mapillary.py

```diff
@@ -94,45 +144,88 @@ def pull_and_save_trace_for_bbox(bbox: str) -> None:
         print('ERROR: Failed to pull trace data: {}'.format(repr(e)))
 
 
-def make_trace_data_requests(session_: requests.Session, bbox: str, conf: any) -> dict[str, list]:
-    result, elapsed = {}, 0
-    start = time.time()
+def make_trace_data_requests(session_: requests.Session, bbox: str, conf: any) -> list[list[dict]]:
+    """
+    Makes the actual calls to Mapillary API to pull trace data for a given bbox string.
+
+    :param session_: requests.Session() to persist session across API calls
+    :param bbox: String representation of bbox that Mapillary API understands, i.e. 'min_lon,min_lat,max_lon,max_lat'
+    :param conf: Dict of configs. Mandatory keys are ['client_id']. Optional keys are ['sequences_per_page',
+        'skip_if_fewer_images_than', 'start_date']
+    :return: List of trace data sequences. Trace data is in format understood by Valhalla map matching process, i.e. it
+        has 'lon', 'lat', 'time', and optionally 'radius' keys
+    """
+    bbox_as_list = [float(d) for d in bbox.split(',')]
+
+    # We will use this dict to group trace points by sequence ID
+    sequences_by_id = {}
 
     map_client_id = conf['client_id']  # The Mapillary client ID, mandatory key of conf
 
     # Check to see if user specified any overrides in conf JSON
     seq_per_page = conf['sequences_per_page'] if 'sequences_per_page' in conf else SEQUENCES_PER_PAGE_DEFAULT
+    img_per_page = conf['images_per_page'] if 'images_per_page' in conf else IMAGES_PER_PAGE_DEFAULT
     skip_if_fewer_imgs_than = conf[
         'skip_if_fewer_images_than'] if 'skip_if_fewer_images_than' in conf else SKIP_IF_FEWER_IMAGES_THAN_DEFAULT
+    start_date = conf['start_date'] if 'start_date' in conf else SEQUENCE_START_DATE_DEFAULT
+
```

Comment:
The logic in this function hasn't changed, I just renamed `sequence` to `seq` to be more consistent so the diff looks large. The "real" changes are highlighted below.